### ISSUE TITLE
About model's result on cropped pictures from CelebA dataset

### ISSUE THREAD

**Issue by ZhaoWeicheng98** [jingyuanli001/RFR-Inpainting#26]
First of all, thank you for your work!

I tried the CelebA pretrained weight on some pictures that were cropped from pictures in the CelebA dataset. These cropped pictures are mostly of faces. However, I found that though the model worked well on pictures originally from the CelebA dataset, it did not give out a good result when cropped pictures were provided. Masks are randomly added to my tests.

For example, here are 2 pictures from the original CelebA dataset.
![000006](https://user-images.githubusercontent.com/35715683/99902891-df7c4500-2cb8-11eb-9959-c576ff490e65.jpg)
![000007](https://user-images.githubusercontent.com/35715683/99902894-e0ad7200-2cb8-11eb-95c8-2bf2d7146219.jpg)

here are the masked of these 2 pictures
![masked_img_4](https://user-images.githubusercontent.com/35715683/99902927-06d31200-2cb9-11eb-9037-5b3adc0a2140.png)
![masked_img_5](https://user-images.githubusercontent.com/35715683/99902928-089cd580-2cb9-11eb-8345-08f57d58609f.png)

The model's results on these 2 pictures are indeed good.
![img_4](https://user-images.githubusercontent.com/35715683/99902960-2ff3a280-2cb9-11eb-8ce7-c5de3f322f9a.png)
![img_5](https://user-images.githubusercontent.com/35715683/99902963-3255fc80-2cb9-11eb-88d0-88f2b143661f.png)

Here are my cropped version of those 2 pictures. I chose the face parts and cropped them out.
![face000006](https://user-images.githubusercontent.com/35715683/99902899-e7d48000-2cb8-11eb-9005-a5d02d922d73.jpg)
![face000007](https://user-images.githubusercontent.com/35715683/99902901-e86d1680-2cb8-11eb-80d7-cc371eba18ff.jpg)

here are the masked of these 2 cropped pictures
![masked_img_14](https://user-images.githubusercontent.com/35715683/99902990-629d9b00-2cb9-11eb-816f-7995d9f4e787.png)
![masked_img_15](https://user-images.githubusercontent.com/35715683/99902997-67624f00-2cb9-11eb-8d1f-4a3557c531a5.png)

However,the results this time are not very good, even they were masked to a similar extent.
![img_14](https://user-images.githubusercontent.com/35715683/99903013-7b0db580-2cb9-11eb-8e3f-3ba5dba96267.png)
![img_15](https://user-images.githubusercontent.com/35715683/99903011-79dc8880-2cb9-11eb-9e5d-50fbe2cf7199.png)

Considering the result, the model might be overfitting? Looking forward to your reply on this test.

**Comment by jingyuanli001** [comment]
Hi, thanks for your question first!
For this question, I thought it is due to the biases in the training images in the dataset, i.e. they put the face in the center of of an image. Then the model might only learn to inpaint images under this setting. Besides, it seems that the cropped images are also resized to to 256, under which setting the model might fail due to the change of context resolution. Can you try to inpaint on the cropped images without resizing them? I thought there should have some way to generate good results even if the images are cropped.

**Comment by ZhaoWeicheng98** [comment]
Thank you for your quick reply.
I tried setting the target size to 128*128 on the cropped images following your instruction. However, it seems better than before, still not very well. 

Masked:
![masked_img_14](https://user-images.githubusercontent.com/35715683/99977046-19724780-2d9c-11eb-9cce-347c1580208a.png)
![masked_img_15](https://user-images.githubusercontent.com/35715683/99976971-03fd1d80-2d9c-11eb-8bba-3af90e662b20.png)

Result:
![img_14](https://user-images.githubusercontent.com/35715683/99976984-06f80e00-2d9c-11eb-9c6a-1a741afee472.png)
![img_15](https://user-images.githubusercontent.com/35715683/99977036-17a88400-2d9c-11eb-8654-10a7976b5071.png)

If the set target size is 128*128 for original CelebA images, the results are worse than before.

Masked:
![masked_img_4](https://user-images.githubusercontent.com/35715683/99977288-58a09880-2d9c-11eb-8c53-285b8ccea0ae.png)ked
![masked_img_5](https://user-images.githubusercontent.com/35715683/99977289-59392f00-2d9c-11eb-8cf9-8eb21920d6a5.png)

Result:
![img_4](https://user-images.githubusercontent.com/35715683/99977277-563e3e80-2d9c-11eb-8ed2-0fa387b274cd.png)
![img_5](https://user-images.githubusercontent.com/35715683/99977286-58080200-2d9c-11eb-8686-3a5145bf9c86.png)

Maybe a re-train or finetune is needed for a cross-resolution generation?
Thanks!

**Comment by jingyuanli001** [comment]
> Thank you for your quick reply.
> I tried setting the target size to 128*128 on the cropped images following your instruction. However, it seems better than before, still not very well.
> 
> Masked:
> ![masked_img_14](https://user-images.githubusercontent.com/35715683/99977046-19724780-2d9c-11eb-9cce-347c1580208a.png)
> ![masked_img_15](https://user-images.githubusercontent.com/35715683/99976971-03fd1d80-2d9c-11eb-8bba-3af90e662b20.png)
> 
> Result:
> ![img_14](https://user-images.githubusercontent.com/35715683/99976984-06f80e00-2d9c-11eb-9c6a-1a741afee472.png)
> ![img_15](https://user-images.githubusercontent.com/35715683/99977036-17a88400-2d9c-11eb-8654-10a7976b5071.png)
> 
> If the set target size is 128*128 for original CelebA images, the results are worse than before.
> 
> Masked:
> ![masked_img_4](https://user-images.githubusercontent.com/35715683/99977288-58a09880-2d9c-11eb-8c53-285b8ccea0ae.png)
> ![masked_img_5](https://user-images.githubusercontent.com/35715683/99977289-59392f00-2d9c-11eb-8cf9-8eb21920d6a5.png)
> 
> Result:
> ![img_4](https://user-images.githubusercontent.com/35715683/99977277-563e3e80-2d9c-11eb-8ed2-0fa387b274cd.png)
> ![img_5](https://user-images.githubusercontent.com/35715683/99977286-58080200-2d9c-11eb-8686-3a5145bf9c86.png)
> 
> Maybe a re-train or finetune is needed for a cross-resolution generation?
> Thanks!

Yeah, given the results from your case, it then seems to be necessary to at least finetune the model for a while.
